File: dotnet/Gitologist/Types/LogEntry.cs
namespace Gitologist.Types;

public record class LogEntry
{
    public required string Sha { get; init; }
    public required string AbbreviatedSha { get; init; }
    public required string Tree { get; init; }
    public string? Parent { get; init; }
    public required string Author { get; init; }
    public required string Committer { get; init; }
    public required DateTime Date { get; init; }
    public required string Message { get; init; }
}

File: zig/src/types/LogEntry.zig
const std = @import("std");

pub const LogEntry = struct {
    sha: []const u8,
    abbreviated_sha: []const u8,
    tree: []const u8,
    parent: ?[]const u8,
    author: []const u8,
    committer: []const u8,
    date: i64,
    message: []const u8,
};
